src/Aspire.Hosting/api/Aspire.Hosting.cs

```diff
@@ -822,7 +846,26 @@ public sealed partial class ContainerNameAnnotation : IResourceAnnotation
         public required string Name { get { throw null; } set { } }
     }
 
-    public partial class ContainerResource : Resource, IResourceWithEnvironment, IResource, IResourceWithArgs, IResourceWithEndpoints, IResourceWithWaitSupport
+    [System.Diagnostics.DebuggerDisplay("{ValueExpression}")]
+    public partial class ContainerPortReference : IManifestExpressionProvider, IValueWithReferences
+    {
+        public ContainerPortReference(IResource resource) { }
+
+        public System.Collections.Generic.IEnumerable<object> References { get { throw null; } }
+
+        public IResource Resource { get { throw null; } }
+
+        public string ValueExpression { get { throw null; } }
+    }
+
+    public partial class ContainerRegistryReferenceAnnotation : IResourceAnnotation
```

Comment:
Should this have `[Experimental("ASPIRECOMPUTE001", UrlFormat = "https://aka.ms/aspire/diagnostics/{0}")]` on it? It's only property - IContainerRegistry does.

Reply:
https://github.com/dotnet/aspire/pull/9285